pytorch_lightning/core/lightning.py

```diff
@@ -1396,7 +1396,9 @@ def _load_model_state(cls, checkpoint: Dict[str, Any]) -> 'LightningModule':
 
         if cls_takes_hparams:
             if ckpt_hparams is not None:
-                is_namespace = checkpoint.get('hparams_type') == 'namespace'
+                is_namespace = False
+                if 'hparams_type' not in checkpoint or checkpoint.get('hparams_type') == 'namespace':
+                    is_namespace = True
```

Comment:
```suggestion
                is_namespace = checkpoint.get('hparams_type', 'namespace') == 'namespace'
```

Reply:
Thanks @Borda for your neat suggestion, I have updated the PR.

Reply:
still @amoudgl pls update changelog... then we can go...